### PR TITLE
Couple command line options are now clamped or checked before they're set

### PR DESCRIPTION
**Pull request by rszczers**
Tried to tackle #111

In particular, it turned out that `--fontDir` was parsed incorrectly. 
```
option --fontDir: cannot parse value `/tmp'
```

### CODE REVIEW

File: engine-src/Game/LambdaHack/Server/Commandline.hs [hunk 1]
```diff
@@ -291,10 +293,16 @@ maxFpsP = optional $ max 1 <$>
...
-              <> metavar "N"
-              <> help "Log only messages of priority at least N, where 1 (all) is the lowest and 5 (errors only) is the default." )
+logPriorityP = optional $
+  option (auto >>= verifyLogPriority) $
+       long "logPriority"
+    <> metavar "N"
+    <> help "Log only messages of priority at least N, where 1 (all) is the lowest and 5 (errors only) is the default."
+  where
+    verifyLogPriority n = do
+      if (n >= 1 && n <= 5)
```
Comment (Mikolaj) [inline engine-src/Game/LambdaHack/Server/Commandline.hs]: Uh, embarrassingly, I use a hacky priority 0 at

 https://github.com/LambdaHack/LambdaHack/blob/v0.9.5.0/test/test.hs#L17

and I fail to cover that in the documentation of the option. Could you include fix the option doc and also allow the hacky 0 value? Thank you.

Comment (rszczers) [reply engine-src/Game/LambdaHack/Server/Commandline.hs]: Oh, I entirely forgot to run tests before making a push :-)

**Comment by rszczers** [comment]
I've imported also `defaultMaxFps`. That's kind of messy I guess, but for now I've no idea where it would be best to factor it out.

**Comment by Mikolaj** [comment]
Well done. Good idea with the `defaultMaxFps`. One last detail: CI indicates it needs to be exported from `Game.LambdaHack.Client.UI.Frontend` (not only with EXPOSE_INTERNAL) and the convention https://github.com/LambdaHack/LambdaHack/blob/v0.9.5.0/LambdaHack.cabal#L34 requires (but sadly it's not checked automatically) that `defaultMaxFps` be imported into `Commandline.hs` by way of `Game.LambdaHack.Client`, and not directly, as it is now. Otherwise it looks perfect.

**Comment by Mikolaj** [comment]
Hmm, now that I think about it, showing the default for `--maxFps` may not be the best idea, because it may be modified in config file and then the value used when `--maxFps` is unspecified differs from the default we show. And showing the value from config file as "default" is a bit misleading, too, because it's a value customized by the user, not an official default. What do you think?

**Comment by rszczers** [comment]
So, as far as I understand, the point of [`defaultMaxFps`](https://github.com/LambdaHack/LambdaHack/blob/1a806394cd472aef8061f0e26c1986eadc5b95be/engine-src/Game/LambdaHack/Client/UI/Frontend.hs#L147) is to initially limit fps in case when parsing of config file goes wrong or if there's no config at all. You're right, nothing good comes from showing that value as default. I'll get rid of this. But how about moving that value (and perhaps more constants such like this one) to [`defClientOptions`](https://github.com/LambdaHack/LambdaHack/blob/93d659e65b519d8f8818b7a0744f2a3588d85ae9/engine-src/Game/LambdaHack/Client/ClientOptions.hs#L77)? Sorry if I'm missing something here, but I don't get the point of current  [`defClientOptions`](https://github.com/LambdaHack/LambdaHack/blob/93d659e65b519d8f8818b7a0744f2a3588d85ae9/engine-src/Game/LambdaHack/Client/ClientOptions.hs#L77). It's part of [`emptyStateClient`](https://github.com/LambdaHack/LambdaHack/blob/147a8c259c27ca008064447283abd2dc1bfe1584/engine-src/Game/LambdaHack/Client/State.hs#L121) which already looks quite arbitrary.

**Comment by Mikolaj** [comment]
> So, as far as I understand, the point of [`defaultMaxFps`](https://github.com/LambdaHack/LambdaHack/blob/1a806394cd472aef8061f0e26c1986eadc5b95be/engine-src/Game/LambdaHack/Client/UI/Frontend.hs#L147) is to initially limit fps in case when parsing of config file goes wrong or if there's no config at all.

Yes, indeed. And actually it's impossible that config file is not parsed, because TH parses a default one and it has the max FPS defined. So that's probably a leftover and technical debt and `fromMaybe defaultMaxFps` should be replaced by `fromJust`, as ugly partial as it looks. So perhaps that particular field should not be a `Maybe`, which moves the `fromJust` closer to the place where the config file is parsed. 

I guess that, and more, should be done as part of the #116 redesign. And actually your further remarks show you spotted the technical debt also in other places. Ugh, it doesn't even have good comments, so I have to reconstruct from memory.

> [`defClientOptions`](https://github.com/LambdaHack/LambdaHack/blob/93d659e65b519d8f8818b7a0744f2a3588d85ae9/engine-src/Game/LambdaHack/Client/ClientOptions.hs#L77)

Currently this is not a record of true default values, but of the default values *before* parsing the config file and commandline.

> It's part of [`emptyStateClient`](https://github.com/LambdaHack/LambdaHack/blob/147a8c259c27ca008064447283abd2dc1bfe1584/engine-src/Game/LambdaHack/Client/State.hs#L121) which already looks quite arbitrary.

Indeed, because it's basically a placeholder to initialize the State monad that holds the client state. Then, IIRC already inside the monad, every thing is parsed, randomly generated, loaded from savegame files, whatever, and only then the state makes sense, in particular, getting some real default values in it. So, I guess it's more `zeroClientState` and `zeroClientOptions` --- noop ones, rather than any meaningful defaults. Feel free to rename if you come up with some consistent scheme. Fixing #116 first may be a good idea, but then it already blocks things for years. :)

**Comment by rszczers** [comment]
Thanks for clarification, somehow I overlooked #116. Ok, let's leave it as it is for now. It's not much, but I need some more time to get my head around all these nuances :-)

**Comment by Mikolaj** [comment]
Splendid. Once CI passes, could you rebase and merge and then close the main issue? I think github now knows you the required authority to do that. Thank you.

**Pull request merged**